files/zh-cn/web/javascript/reference/global_objects/generator/throw/index.md

```diff
@@ -1,54 +1,69 @@
 ---
 title: Generator.prototype.throw()
 slug: Web/JavaScript/Reference/Global_Objects/Generator/throw
+l10n:
+  sourceCommit: fb85334ffa4a2c88d209b1074909bee0e0abd57a
 ---
 
 {{JSRef}}
 
-**`throw()`** 方法用来向生成器抛出异常，并恢复生成器的执行，返回带有 `done` 及 `value` 两个属性的对象。
+{{jsxref("Generator")}} 实例的 **`throw()`** 方法的作用就好像一个 `throw` 语句被插入到生成器主体的当前暂停位置，这会通知生成器错误的情况并允许其处理错误，或者执行清理和自行关闭。
 
 ## 语法
 
-```plain
-gen.throw(exception)
+<!-- 我们通常不会为 return 添加“generatorInstance”主体。然而，这里是必要的，因为“throw”是一个关键字，所以否则它是无效的语法。 -->
```

Comment:
Let's remove this, see: https://github.com/mdn/translated-content/pull/17711#discussion_r1441219843